pkg/client/unversioned/helper.go

```diff
@@ -184,7 +186,7 @@ func MatchesServerVersion(client *Client, c *Config) error {
 //   stderr and try client's registered versions in order of preference.
 // - If version is config default, and the server does not support it,
 //   return an error.
-func NegotiateVersion(client *Client, c *Config, version string, clientRegisteredVersions []string) (string, error) {
+func NegotiateVersion(client *Client, c *Config, group string, version string, clientRegisteredGroupVersions []string) (string, error) {
 	var err error
 	if client == nil {
 		client, err = New(c)
```

Comment:
NegotiateVersion currently requires a default client.Client to communicate with the server, if the client is nil, NegotiateVersion will generate one based on the config. This approach seems wrong to me, because we can't know what version the server supports without doing the negotiation first.
 I plan to create a RESTClient with the config, but ignoring the group and version in the config, and use this unversioned RESTClient to communicate with the server during the negotiation.